packages/qwik/src/core/container/serializers.ts

```diff
@@ -442,6 +442,22 @@ const BigIntSerializer = /*#__PURE__*/ serializer<bigint>({
   },
 });
 
+const decoder = new TextDecoder();
+const Uint8ArraySerializer = /*#__PURE__*/ serializer<Uint8Array>({
+  $prefix$: '\u001c',
+  $test$: (v) => v instanceof Uint8Array,
+  $serialize$: (v) => btoa(decoder.decode(v)),
+  $prepare$: (data) => {
+    const buf = atob(data);
+    const array = new Uint8Array(buf.length);
+    for (let i = 0; i < buf.length; i++) {
+      array[i] = buf.charCodeAt(i);
+    }
+    return array;
```

Comment:
Why don't you use TextEncoder here?

Reply:
Sorry, I recalled the TextEncoder and TextDecoder is not usable for this purpose.
The TextDecoder can decode only the encoded array of text.
It is need to use the generic base64 encoder for any binary data.